internal/app/machined/pkg/controllers/k8s/kubelet_service.go

```diff
@@ -201,6 +212,17 @@ func (ctrl *KubeletServiceController) Run(ctx context.Context, r controller.Runt
 	}
 }
 
+// extractHostname extracts the hostname from the kubelet args.
+func (ctrl *KubeletServiceController) extractHostname(kubeletArgs []string) (string, error) {
```

Comment:
i wonder what happens if a user set this to empty?

Reply:
Good point. It worked, but it was removing certs each time controller is run because of hostname mismatch each time. I changed it to return the system hostname as fallback now, which avoids recreating the certs when `--hostname-override=""`.